src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -233,7 +236,39 @@ else if (pathResult.getStatus() == PathFindingStatus.CALCULATION_COMPLETE)
         {
             return;
         }
-        super.tick();
+
+        ++this.totalTicks;
+        if (this.tryUpdatePath)
+        {
+            this.updatePath();
+        }
+
+        // The following block replaces mojangs super.tick(). Why you may ask? Because it's broken, that's why.
+        if (!this.noPath())
+        {
+            if (this.canNavigate())
+            {
+                this.pathFollow();
+            }
+            else if (this.currentPath != null && !this.currentPath.isFinished())
+            {
+                Vector3d vector3d = this.getEntityPosition();
+                Vector3d vector3d1 = this.currentPath.getPosition(this.entity);
+                if (vector3d.y > vector3d1.y && !this.entity.isOnGround() && MathHelper.floor(vector3d.x) == MathHelper.floor(vector3d1.x) && MathHelper.floor(vector3d.z) == MathHelper.floor(vector3d1.z))
+                {
+                    this.currentPath.incrementPathIndex();
+                }
+            }
+
+            DebugPacketSender.sendPath(this.world, this.entity, this.currentPath, this.maxDistanceToWaypoint);
+            if (!this.noPath())
+            {
+                Vector3d vector3d2 = this.currentPath.getPosition(this.entity);
+                BlockPos blockpos = new BlockPos(vector3d2);
+                this.entity.getMoveHelper().setMoveTo(vector3d2.x, this.world.getBlockState(blockpos.down()).isAir() ? vector3d2.y : getSmartGroundY(this.world, blockpos), vector3d2.z, this.speed);
+            }
+        }
+        // End of super.tick.
```

Comment:
More documentation on why it is broken in case it changes in the future is needed here.
Also I would suggest documenting somehow how to test this.

Reply:
done